### PR TITLE
4 setup project structure

### PR DESCRIPTION
**Pull request by abedshamia**
Setup the project (file structure , Eslint, packages and Server setup)

- to install packages run ``` yarn install ```

### CODE REVIEW

**Review by manar-abed — APPROVED** [review]

File: .eslintrc.json [hunk 1]
```diff
@@ -0,0 +1,18 @@
+{
+  "env": {
+    "browser": true,
+    "es2021": true,
+    "node": true
+  },
```
Comment (Raghad-Mezied) [inline .eslintrc.json:6]: This `eslint` is for your backend side, where there is a browser env here?

File: client/README.md [hunk 2]
```diff
@@ -0,0 +1,70 @@
+# Getting Started with Create React App
+
+This project was bootstrapped with [Create React App](https://github.com/facebook/create-react-app).
```
Comment (Raghad-Mezied) [inline client/README.md:3]: Delete this file we don't need it

File: client/package.json [hunk 3]
```diff
@@ -0,0 +1,39 @@
+{
+  "name": "client",
+  "version": "0.1.0",
+  "private": true,
+  "dependencies": {
+    "@testing-library/jest-dom": "^5.14.1",
+    "@testing-library/react": "^13.0.0",
+    "@testing-library/user-event": "^13.2.1",
```
Comment (Raghad-Mezied) [inline client/package.json:8]: Remove testing libraries, install antd and redux.

File: client/public/index.html [hunk 4]
```diff
@@ -0,0 +1,40 @@
+<!DOCTYPE html>
+<html lang="en">
+  <head>
+    <meta charset="utf-8" />
+    <link rel="icon" href="%PUBLIC_URL%/favicon.ico" />
+    <meta name="viewport" content="width=device-width, initial-scale=1" />
+    <meta name="theme-color" content="#000000" />
+    <meta name="description" content="Web site created using create-react-app" />
+    <link rel="apple-touch-icon" href="%PUBLIC_URL%/logo192.png" />
+    <!--
```
Comment (Raghad-Mezied) [inline client/public/index.html:10]: Add your logo as an icon here and delete all files and images that we don't need.

File: package.json [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    "ts-node": "^10.7.0",
+    "typescript": "^4.6.3"
+  },
+  "devDependencies": {
+    "@types/express": "^4.17.13",
+    "cross-env": "^7.0.3",
+    "eslint": "^8.14.0",
+    "jest": "^28.0.1",
+    "nodemon": "^2.0.15",
+    "supertest": "^6.2.2"
```
Comment (Raghad-Mezied) [inline package.json:35]: Are you sure that installed typescript?

File: server/app.ts [hunk 6]
```diff
@@ -0,0 +1,5 @@
+import express, {Application} from 'express';
+
+const app: Application = express();
+
```
Comment (Raghad-Mezied) [inline server/app.ts:4]: Use all required middlewares.

File: server/server.ts [hunk 7]
```diff
@@ -0,0 +1,7 @@
+import app from './app';
+
+const PORT: number = Number(process.env.PORT) || 3001;
+
+app.listen(PORT, () => {
+  console.log(`Server is running on port ${PORT}`);
```
Comment (Raghad-Mezied) [inline server/server.ts:6]: we are already inside a server folder do just name it index.ts

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

File: client/src/index.js [hunk 8]
```diff
@@ -0,0 +1,12 @@
+import React from 'react';
+import ReactDOM from 'react-dom/client';
+import './index.css';
+import App from './App';
+import reportWebVitals from './reportWebVitals';
```
Comment (Raghad-Mezied) [inline client/src/index.js:5]: You deleted this file, so remove all imports of deleted files, delete all unneeded code, and make sure the app is running without problems.

**Review by Raghad-Mezied — CHANGES_REQUESTED** [review]

**Review by Mahmoud-Ahmad2 — APPROVED** [review]

**Review by Raghad-Mezied — APPROVED** [review]

**Pull request merged**